x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/create_engine_flyout.tsx

```diff
@@ -0,0 +1,198 @@
+                enginesDocsLink: (
+                  <EuiLink
+                    href="#" // TODO: replace with docs link
+                    target="_blank"
+                    data-telemetry-id="entSearchContent-engines-createEngine-docsLink"
+                    external
+                  >
+                    {i18n.translate(
+                      'xpack.enterpriseSearch.content.engines.createEngine.header.docsLink',
+                      { defaultMessage: 'Engines documentation' }
+                    )}
+                  </EuiLink>
+                ),
+              }}
+            />
+          </p>
+        </EuiText>
+        {createEngineStatus === Status.ERROR && createEngineError && (
+          <>
+            <EuiSpacer />
+            <EuiCallOut
+              color="danger"
+              title={i18n.translate(
+                'xpack.enterpriseSearch.content.engines.createEngine.header.createError.title',
+                { defaultMessage: 'Error creating engine' }
+              )}
+            >
+              {getErrorsFromHttpResponse(createEngineError).map((errMessage, i) => (
+                <p id={`createErrorMsg.${i}`}>{errMessage}</p>
+              ))}
+            </EuiCallOut>
+          </>
+        )}
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <EuiSteps
+          steps={[
+            {
+              children: (
+                <IndicesSelectComboBox
+                  fullWidth
+                  isDisabled={formDisabled}
+                  onChange={onIndicesChange}
+                  selectedOptions={selectedIndices.map(indexToOption)}
+                />
+              ),
+              status: indicesStatus,
+              title: i18n.translate(
+                'xpack.enterpriseSearch.content.engines.createEngine.selectIndices.title',
+                { defaultMessage: 'Select indices' }
+              ),
+            },
+            {
+              children: (
+                <EuiFieldText
+                  fullWidth
+                  disabled={formDisabled}
+                  placeholder={i18n.translate(
+                    'xpack.enterpriseSearch.content.engines.createEngine.nameEngine.placeholder',
+                    { defaultMessage: 'Engine name' }
+                  )}
+                  value={engineName}
+                  onChange={(e) => setEngineName(e.target.value)}
+                />
+              ),
+              status: engineNameStatus,
+              title: i18n.translate(
+                'xpack.enterpriseSearch.content.engines.createEngine.nameEngine.title',
+                { defaultMessage: 'Name your engine' }
+              ),
+            },
+          ]}
+        />
+      </EuiFlyoutBody>
+      <EuiFlyoutFooter>
+        <EuiFlexGroup>
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              disabled={formDisabled}
+              data-telemetry-id={`entSearchContent-engines-createEngine-cancel`}
```

Comment:
nitpick: just make this a string instead of javascript code returning a string ;) (same for some other `data-telemetry-id`s here